app/workers/search_indexer/tasks.py

```
@@ -0,0 +1,278 @@
+                current_pid,
+            )
+            new_worker = SearchIndexerWorker()
+            await new_worker.initialize()
+            # Присваиваем глобальным переменным только после успешной инициализации
+            worker_instance = new_worker
+            worker_instance_pid = current_pid
+            logger.info(
+                "Search Indexer Worker успешно инициализирован в процессе %s.",
+                current_pid,
+            )
+            return worker_instance
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)